### ISSUE TITLE
clang-17: error: unsupported option '-mfpu=' for target 'aarch64-unknown-linux-musl'

### ISSUE THREAD

**Issue by 12101111** [llvm/llvm-project#74361]
I found this issue from configure script in ghostscript-gpl

```shell
> clang-16 -mfpu=neon test.c
clang-16: warning: argument unused during compilation: '-mfpu=neon' [-Wunused-command-line-argument]
> clang-16 -mfpu=neon test.c -lm
> clang-17 -mfpu=neon test.c
clang-17: error: unsupported option '-mfpu=' for target 'aarch64-unknown-linux-musl'
> clang-17 -mfpu=neon test.c -lm
> gcc -mfpu=neon test.c
gcc: error: unrecognized command-line option '-mfpu=neon'
> gcc -mfpu=neon test.c -lm
gcc: error: unrecognized command-line option '-mfpu=neon'
```

test.c:

```c
#include <arm_neon.h>
int main() {
  int32x4_t round = vdupq_n_s32(10);
  return 0;
}
```

clang 17 has a inconsistent behavior depend on whether a linker flag is given or not.

**Comment by MaskRay** [comment]
This is a missing feature in the clang driver. 

https://maskray.me/blog/2023-08-25-clang-wunused-command-line-argument

> Options that are relevant only during compilation do not result in `-Wunused-command-line-argument` diagnostics when linking is performed.

They also don't lead to "unsupported option" errors when linking is performed.
Certain driver options only affect linking are marked as `LinkerInput` in Clang.

```
% clang --target=aarch64 -mavx a.c -fdriver-only      # should report an error but not
clang: error: unsupported option '-mavx' for target 'aarch64'
% clang --target=aarch64 -mavx a.c b.o -fdriver-only  # should report an error but not
% clang --target=aarch64 -mavx a.c -lm -fdriver-only  # should report an error but not
```

**Comment by brunvonlope** [comment]
A workaround is to disable NEON optimizations with `--disable-neon` option. Anyway, I wonder if ghostscript devs are aware of it
